Pinta.Core/Managers/ToolManager.cs

```diff
@@ -77,7 +79,18 @@ public void RemoveInstanceOfTool (System.Type tool_type)
 				}
 			}
 		}
-		
+
+		public void CreateGoupedTools()
```

Comment:
I think this should be merged into the AddTool() and RemoveTool() methods - it's possible for add-ins to register additional tools, so the `groupedTools` dictionary should be kept up to date in those methods instead of having a separate method to call.

Reply:
That's a great idea! Thanks.